frontend/src/pages/GraphicalSelector/GraphicalSelector.jsx

```diff
@@ -72,6 +91,44 @@ const GraphicalSelector = () => {
       const { _cfg: { id } } = node;
       const [courseData, err] = await axiosRequest("get", `/courses/getCourse/${id}`);
       if (!err) setCourse(courseData);
+
+      // hides/ unhides dependent nodes
+      if (node.hasState("click")) {
+        graphInstance.setItemState(node, "click", false);
+        const { breadthFirstSearch } = Algorithm;
+        breadthFirstSearch(data, id, {
+          enter: ({ current }) => {
+            if (id !== current) {
+              const currentNode = graphInstance.findById(current);
+              // Unhiding node won't unhide other hidden nodes
+              currentNode.getEdges().forEach((e) => e.show());
+              currentNode.show();
+            }
+          },
+        });
+      } else if (node.getOutEdges().length !== 0) {
```

Comment:
`node.getOutEdges().length` instead? Dont need to compare to 0 right? lol

Reply:
what's ligma?

Reply:
🟣💧👄💧🟣